apps/src/templates/teacherDashboard/teacherSectionsRedux.js

```diff
@@ -114,6 +118,45 @@ export const finishEditingSection = () => (dispatch, getState) => {
   });
 };
 
+export const asyncLoadSectionData = () => (dispatch) => {
+  dispatch({type: ASYNC_LOAD_BEGIN});
+  return Promise.all([
+    fetchSections(),
+    fetchValidCourses(),
+    fetchValidScripts()
+  ]).then(([sections, validCourses, validScripts]) => {
+    dispatch(setValidAssignments(validCourses, validScripts));
+    dispatch(setSections(sections));
+    dispatch({type: ASYNC_LOAD_SUCCESS});
+  }).catch(() => {
+    dispatch({type: ASYNC_LOAD_FAILURE});
+  });
+};
+
+function fetchSections() {
+  return new Promise((resolve, reject) => {
+    $.getJSON('/dashboardapi/sections/')
+      .done(resolve)
+      .fail(reject);
+  });
+}
+
+function fetchValidCourses() {
+  return new Promise((resolve, reject) => {
+    $.getJSON('/dashboardapi/courses')
+      .done(resolve)
+      .fail(reject);
+  });
+}
+
+function fetchValidScripts() {
+  return new Promise((resolve, reject) => {
+    $.getJSON('/v2/sections/valid_scripts')
+      .done(resolve)
+      .fail(reject);
+  });
+}
```

Comment:
I'm off the `fetch` polyfill idea for a bit, just because there are some extra steps you're supposed to take with `fetch` to verify a successful request and convert the request body to JSON.  I like what you've proposed here.